### ISSUE TITLE
Clang should warn about chained comparisons like `a > b > c`

### ISSUE THREAD

**Issue by zmodem** [llvm/llvm-project#60256]
We had this `=` vs. `==` bug in Chromium recently:

```c++
int f(int);
int g(int a) {
  a == f(a) > 1 ? 1 : f(a);
  // stuff
}
```

In most cases, Clang would catch a `=` vs `==` bug with `-Wunused-comparison`, but in this case the unintentional comparison became an operand to the ternary operator, and no warning fired.

Perhaps there should be warning for the general case of "chained comparisons" that would catch the above as well as code like:

```c++
void h(int a, int b, int c) {
  if (a > b > c) {
    // stuff...
  }
}
```

**Comment by hazohelet** [comment]
Let me work on this issue.
I guess the change for this would be to add a new warning flag within `-Wparentheses` that detects comparison operators within comparison operators, just like '&&' within '||'.

**Comment by cor3ntin** [comment]
Fixed https://github.com/llvm/llvm-project/commit/f39af73f3edcc949bf9dc3535effc59afbcdda22

**Comment by zmodem** [comment]
Cool!

It looks like the patch doesn't cover the original example with the ternary operator, though. It seems this is because it only considers BinaryOperators. Would it be possible to expand it?
